### PR TITLE
tests: Levenshtein Distance (dynamic programming solution)

### PR DESCRIPTION
**Pull request by JCarlosR**
### Describe your change:

- [x] Documentation change
- [x] Add tests for an existing algorithm 
- [x] Add missing comment block on existing algorithm

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

File: Dynamic-Programming/tests/LevenshteinDistance.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { calculateLevenshteinDp } from '../LevenshteinDistance'
+
+test('Test Case 1', () => {
```
Comment (appgurueu) [inline Dynamic-Programming/tests/LevenshteinDistance.test.js:3]: Please use descriptive test case titles (or merge all tests).

Comment (JCarlosR) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js]: I just updated the titles, please let me know if they are fine

File: Dynamic-Programming/tests/LevenshteinDistance.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { calculateLevenshteinDp } from '../LevenshteinDistance'
+
+test('Add and remove letters at the beginning and the end while making replacements', () => {
```
Comment (CarlosZoft) [inline Dynamic-Programming/tests/LevenshteinDistance.test.js:3]: Suggestion :
"Should return the distance based on the characters in the middle of the strings"

File: Dynamic-Programming/tests/LevenshteinDistance.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const from = 'kitten'
+  const to = 'sitting'
+  expect(calculateLevenshteinDp(from, to)).toBe(3)
+})
+
+test('Replace contiguous letters at the middle of the words', () => {
```
Comment (CarlosZoft) [inline Dynamic-Programming/tests/LevenshteinDistance.test.js:9]: Suggestion :
"Should return distance based on characters at the ends of strings"

File: Dynamic-Programming/tests/LevenshteinDistance.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  const from = 'book'
+  const to = 'back'
+  expect(calculateLevenshteinDp(from, to)).toBe(2)
+})
+
+test('Compare properly strings with different length', () => {
```
Comment (CarlosZoft) [inline Dynamic-Programming/tests/LevenshteinDistance.test.js:15]: this test is redundant, the first test already has different sizes

**Review by CarlosZoft — CHANGES_REQUESTED** [review]

File: Dynamic-Programming/tests/LevenshteinDistance.test.js (hunk 4 above)
Comment (JCarlosR) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js:15]: I believe the more tests the better. Sometimes we see more specific test cases not passing surprisingly.
Simple test cases give us confidence while reviewing the complex ones.

How would you describe this one?

File: Dynamic-Programming/tests/LevenshteinDistance.test.js (hunk 2 above)
Comment (JCarlosR) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js:3]: The test titles on the repository have various formats. 

@appgurueu Please let me know if you also think adding "should return the distance" at the beginning is preferable, or a general title as it is is enough.

Comment (CarlosZoft) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js:3]: @JCarlosR, the importance is that the description of your tests, follow a pattern. For readability :)

File: Dynamic-Programming/tests/LevenshteinDistance.test.js (hunk 4 above)
Comment (CarlosZoft) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js:15]: Okayy ! 

I would put something like :  "Should return distance when strings length are different"

Comment (CarlosZoft) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js:15]: "return" because you are testing a function result

Comment (JCarlosR) [reply Dynamic-Programming/tests/LevenshteinDistance.test.js]: Thanks

**Review by CarlosZoft — APPROVED** [review]
LGTM 🔥

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**